libvips/iofuncs/util.c

```diff
@@ -612,7 +612,7 @@ vips__open( const char *filename, int flags, int mode )
 	fd = g_open( filename, flags, mode );
 
 #ifdef G_OS_WIN32
-	if( mode & O_CREAT )
+	if( fd > 2 && mode & O_CREAT )
```

Comment:
Perhaps we could should move this check to `vips__set_create_time`, since it's also used by `vips__fopen`?
```diff
--- a/libvips/iofuncs/util.c
+++ b/libvips/iofuncs/util.c
@@ -586,7 +586,8 @@ vips__set_create_time( int fd )
 	SYSTEMTIME st;
 	FILETIME ft;
 
-	if( (handle = (HANDLE) _get_osfhandle( fd )) == INVALID_HANDLE_VALUE )
+	if( fd < 3 ||
+		(handle = (HANDLE) _get_osfhandle( fd )) == INVALID_HANDLE_VALUE )
 		return;
 	GetSystemTime( &st );
 	SystemTimeToFileTime( &st, &ft );
```

After checking `vips__fopen`, I think that also needs this patch:
```diff
--- a/libvips/iofuncs/util.c
+++ b/libvips/iofuncs/util.c
@@ -635,7 +635,7 @@ vips__fopen( const char *filename, const char *mode )
 	fp = g_fopen( filename, mode );
 
 #ifdef G_OS_WIN32
-	if( mode[0] == 'w' )
+	if( fp && mode[0] == 'w' )
 		vips__set_create_time( _fileno( fp ) ); 
 #endif /*G_OS_WIN32*/
```
(since `_fileno( NULL );` can crash too, arghh)

Reply:
Done, thanks Kleis.